File: organization_backend/internal/domain/request.go
package domain

import "time"

type Request struct {
	ID                           string         `json:"id"`
	CustomerID                   string         `json:"customerId"`
	DeliveryDate                 time.Time      `json:"deliveryDate"`
	PlannedReturnDate            *time.Time     `json:"plannedReturnDate,omitempty"`
	IntendedStudents             int            `json:"intendedStudents"`
	Status                       string         `json:"status"`
	Archived                     bool           `json:"archived"`
	ApprovedDistributionCenterID *string        `json:"approvedDistributionCenterId,omitempty"`
	OutgoingTrackingCode         *string        `json:"outgoingTrackingCode,omitempty"`
	ShippingCustomerName         string         `json:"shippingName"`
	ShippingAddressLine1         string         `json:"addressLine1"`
	ShippingAddressLine2         string         `json:"addressLine2"`
	ShippingCity                 string         `json:"city"`
	ShippingZipCode              string         `json:"zipCode"`
	Metadata                     map[string]any `json:"metadata"`
	CreatedAt                    time.Time      `json:"createdAt"`
	UpdatedAt                    time.Time      `json:"updatedAt"`
	Items                        []RequestItem  `json:"items"`
}

type RequestItem struct {
	MaterialTypeID string `json:"materialTypeId"`
	Quantity       int    `json:"quantity"`
}

type CreateRequestInput struct {
	CustomerID           string
	DeliveryDate         time.Time
	PlannedReturnDate    time.Time
	IntendedStudents     int
	ShippingCustomerName string
	ShippingAddressLine1 string
	ShippingAddressLine2 string
	ShippingCity         string
	ShippingZipCode      string
	Note                 string
	Items                []RequestItem
}

File: frontend/user/src/types/request.ts
export interface RequestItem {
  materialTypeId: string;
  quantity: number;
}

export interface Request {
  id: string;
  customerId: string;
  deliveryDate: string;
  plannedReturnDate?: string;
  intendedStudents: number;
  status: 'pending' | 'approved' | 'inAction' | 'returned' | 'cancelled';
  outgoingTrackingCode?: string;
  shippingName: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  zipCode: string;
  metadata: Record<string, unknown>;
  createdAt: string;
  updatedAt: string;
  items: RequestItem[];
}

export interface CreateRequestInput {
  deliveryDate: string;
  plannedReturnDate: string;
  intendedStudents: number;
  dataProcessingConsent: boolean;
  shareIntendedStudents: boolean;
  shippingName: string;
  addressLine1: string;
  addressLine2?: string;
  city: string;
  zipCode: string;
  note?: string;
  items: RequestItem[];
}
